ui/src/views/tools/wordanalyser/textupload/TextUpload.js

```diff
@@ -0,0 +1,75 @@
+import React, {useState} from 'react';
+import TextUploadModal from "./TextUploadModal";
+import "../styles/TextUpload.css";
+import {Button, Grid} from "@mui/material";
+import FileUploadIcon from '@mui/icons-material/FileUpload';
+
+function TextUpload(props) {
+    const [buttonPopup, setButtonPopup] = useState(false);
+
+    function fileUpload() {
+        let data_input = document.getElementById("form_data");
+        let oData = new FormData(data_input);
+        const request_test = {
+            method: "POST",
+            body: oData
+        }
+
+        fetch("/api/textfromfile", request_test)
+        .then(response => response.text())
+        .then(data => {
+            props.sendTextFromFile(data);
+        });
+        document.getElementById("file_name").innerHTML = "";
+    }
+
+    function filechange() {

```

Comment:
Reactis tavaliselt ei läheneta elementidele id-de abil. Võiks võtta kas eventi küljest target või siis kasutada ref-e. https://reactjs.org/docs/refs-and-the-dom.html